Lib/subprocess.py

```diff
@@ -869,6 +869,27 @@ def pipe_cloexec(self):
 
 
         def _close_fds(self, but):
+            if sys.platform.startswith("linux"):
+                # Try closing the actual opened fds (obtained via /proc),
+                # as this is way faster than iterating through MAX_FD
+                try:
+                    for f in os.listdir("/proc/self/fd"):
+                        try:
+                            fd = int(f)
+                        except:
```

Comment:
Please be precise on the expected exception type.
`except TypeError` here maybe?

Reply:
Mostly `int()` returns `ValueError`, but sometimes also `TypeError`, so I guess both.